fractions/keycloak-microprofile-jwt/pom.xml

```diff
@@ -0,0 +1,62 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright 2015 Red Hat, Inc. and/or its affiliates.
+  ~
+  ~ Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+  -->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>io.thorntail</groupId>
+    <artifactId>build-parent</artifactId>
+    <version>2.3.1.Final-SNAPSHOT</version>
+    <relativePath>../../build-parent/pom.xml</relativePath>
+  </parent>
+
+  <groupId>io.thorntail</groupId>
+  <artifactId>keycloak-microprofile-jwt</artifactId>
+
+  <name>Keycloak Microprofile JWT</name>
+  <description>Securing RESTful services with Keycloak and MP JWT</description>
+
+  <packaging>jar</packaging>
+
+  <properties>
+    <swarm.fraction.stability>stable</swarm.fraction.stability>
+    <swarm.fraction.tags>Security,JWT,Web</swarm.fraction.tags>
+  </properties>
+
+  <build>
+    <resources>
+      <resource>
+        <directory>src/main/resources</directory>
+        <filtering>true</filtering>
+      </resource>
+    </resources>
+  </build>
+
+  <dependencies>
+    <dependency>
+      <groupId>io.thorntail</groupId>
+      <artifactId>microprofile-jwt</artifactId>
+    </dependency>
```

Comment:
Shouldn't it also depend on `io.thorntail:keycloak`?

Reply:
No, keycloak fraction is using a different mechanism, it needs a KEYCLOAK login type, and uses the undertow adapter code to validate. The new fraction works with the Mp JWT mandated MP-JWT login type and the KC factory shipped inside it uses a simpler mechanism for validating because it is constrained by the microprofile-jwt fraction auth mechanism implementation, i.e it only returns 401 if the token validation fails. So in the future I'd also consider updating the new fraction to make this authentication mechanism to be based on the same underlying undertow KC adapter code as the one the 'keycloak' fraction is based upon, but it is a next step :-)

Reply:
So the user must not depend on both of these fractions (`keycloak` and `keycloak-microprofile-jwt`) simultaneously?

Reply:
They won't conflict if they do, or if for some reasons they need to support both MP-JWT and KEYCLOAK login types. It is just a 'keycloak' fraction has no effect on the microprofile-jwt flow...